diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 148e913..a698067 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -798,6 +798,7 @@ config("compiler_cpu_abi") {
         ldflags += [ "--target=aarch64-linux-gnu" ]
       }
     } else if (current_cpu == "mipsel" && !is_nacl) {
+      cflags += [ "-mnan=legacy" ]
       ldflags += [ "-Wl,--hash-style=sysv" ]
       if (custom_toolchain == "") {
         if (is_clang) {
@@ -837,6 +838,29 @@ config("compiler_cpu_abi") {
             "-mfp64",
           ]
         }
+      } else if (mips_arch_variant == "r5") {
+        ldflags += [ "-mips32r5" ]
+        if (is_clang) {
+          cflags += [
+            "-march=mipsel",
+            "-mcpu=mips32r5",
+          ]
+        } else {
+          cflags += [
+            "-mips32r5",
+            "-Wa,-mips32r5",
+          ]
+          if (mips_float_abi == "hard" && mips_fpu_mode != "") {
+            cflags += [ "-m$mips_fpu_mode" ]
+          }
+        }
+        if (mips_use_msa == true) {
+          cflags += [
+            "-mmsa", "-flax-vector-conversions",
+            "-mfp64",
+            "-mno-odd-spreg",
+          ]
+        }
       } else if (mips_arch_variant == "r2") {
         ldflags += [ "-mips32r2" ]
         if (is_clang) {

